### ISSUE TITLE
helm push defaulting to oci://

### ISSUE THREAD

**Issue by gillesde** [chartmuseum/helm-push#109]
## Description

Helm Push issue since Helm v3.7.0 was released on September 15th 2021:
`Error: scheme prefix missing from remote (e.g. "oci://")`

## Steps

```
# Install latest Helm
curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 > get_helm.sh
chmod 700 get_helm.sh
./get_helm.sh --version v3.7.0

# Install helm-push
helm plugin install https://github.com/chartmuseum/helm-push

# Run chart museum locally (tried various versions)
docker run -d -e DEBUG=true -p 8080:8080 -e STORAGE=local -e STORAGE_LOCAL_ROOTDIR=/tmp chartmuseum/chartmuseum:latest

# Add local repo
helm repo add local-development http://localhost:8080
```

## Problem Details

Getting an error OOTB
```
$ helm push mychart/ local-development 
Error: this feature has been marked as experimental and is not enabled by default. Please set HELM_EXPERIMENTAL_OCI=1 in your environment to use this feature
```

Setting the env variable `HELM_EXPERIMENTAL_OCI` results in the issue (tried value 0 and 1)
```
$ export HELM_EXPERIMENTAL_OCI=1
$ helm push mqserver/ local-development 

Error: scheme prefix missing from remote (e.g. "oci://")
```

**Comment by tomjohnburton** [comment]
Yeah I'm getting this too!

**Comment by thcdrt** [comment]
Same here

**Comment by thcdrt** [comment]
In the Helm 3.7.0 [changelog](https://github.com/helm/helm/releases) there is a change that may be the cause of this:

> helm chart push has been replaced with helm push

**Comment by gillesde** [comment]
That's right, the `helm push` command might be integrated within helm now.

But when I tried without installing the `helm-push` plugin, I am still getting the same issue.

Attempted to use the Docker registry image instead of ChartMuseum, which seems to be [recommended by the documentation](https://helm.sh/docs/topics/registries/), but getting another error:
```
$ htpasswd -cB -b auth.htpasswd myuser mypass


$ docker run -dp 5000:5000 --restart=always --name registry \
    -v $(pwd)/auth.htpasswd:/etc/docker/registry/auth.htpasswd \
    -e REGISTRY_AUTH="{htpasswd: {realm: localhost, path: /etc/docker/registry/auth.htpasswd}}" \
    registry


$ export HELM_EXPERIMENTAL_OCI=1


$ helm registry login localhost:5000 -u myuser -p mypass

WARNING: Using --password via the CLI is insecure. Use --password-stdin.
INFO[0000] Error logging in to v2 endpoint, trying next endpoint: Get "https://localhost:5000/v2/": http: server gave HTTP response to HTTPS client 
Login Succeeded


$ helm push helm-charts/test-1.2.3.tgz oci://local-development 
Error: failed to do request: Head "https://local-development/v2/test/blobs/sha256:f0f0ae4641389710015a6b4367828d54ff4d76c98062204cad4cdaf69063aa86": dial tcp: lookup local-development on x.x.x.:53: no such host
```

**Comment by c0dearm** [comment]
Same issue for me, when doing `helm push` with the latest Helm version it now used the built-in command, instead of the ChartMuseum `helm-push`, which doesn't work with ChartMuseum registries like Harbor.

We need a new release of this plugin which doesn't collide with the built-in command to have it working again.

**Comment by devinnasar** [comment]
I wrote a small script with contacts the ChartMuseum API directly as a workaround:

```bash
#!/bin/bash

# Requires:
# yq: https://github.com/mikefarah/yq
# SOPS: https://github.com/mozilla/sops

# This script presumes your ChartMuseum credentials are encrypted via SOPS as YAML in the following format:
#
# credentials.enc.yaml
#
# credentials:
#   chartmuseum:
#     username: "XXX"
#     password: "YYY"

# Self awareness
_script_path="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
_script_working_directory="${_script_path}/working"

# Constants
_chartmuseum_url="https://chartmuseum.example.org/api/charts"

# Inputs
_sops_file_path="${1}"
_helm_chart_path="${2}"

# Functions
_setup(){
    mkdir -p "${_script_working_directory}"
}

_teardown(){
    rm -rf "${_script_working_directory}"
}
trap _teardown EXIT

_main(){
    echo "SOPS file path: ${_sops_file_path}"
    echo "Helm Chart path: ${_helm_chart_path}"

    _cm_user="$(sops --decrypt --extract '["credentials"]["chartmuseum"]["username"]' "${_sops_file_path}")"
    _cm_pass="$(sops --decrypt --extract '["credentials"]["chartmuseum"]["password"]' "${_sops_file_path}")"

    helm package "${_helm_chart_path}" --destination "${_script_working_directory}"

    _package_name="$(yq e '.name' "${_helm_chart_path}/Chart.yaml")"
    _package_version="$(yq e '.version' "${_helm_chart_path}/Chart.yaml")"
    _package_path="${_script_working_directory}/${_package_name}-${_package_version}.tgz"

    # Delete an existing package of the current version prior to upload (mimicking --force)
    curl -X DELETE -u "${_cm_user}:${_cm_pass}" "${_chartmuseum_url}/${_package_name}/${_package_version}"

    # Upload the package
    curl --data-binary "@${_package_path}" -u "${_cm_user}:${_cm_pass}" "${_chartmuseum_url}"
}

_main
```
Feel free to modify as needed.

**Comment by emanuelflp** [comment]
Guys, I opened an PR trying to solve this issue.

It works for me, so maybe you can use: https://github.com/chartmuseum/helm-push/pull/110

To install my plugin version(that is the same of official, just changing the caller command), just install from my repo.

```bash
helm plugin install https://github.com/emanuelflp/helm-push.git
```

**Comment by maxisam** [comment]
> And change the command from `helm push` to `helm legacy-push`

This is awesome! Thank you! on naming, do you think `cm-push` might fit to its meaning more? That said, naming is the hardest problem in computer science lol

**Comment by eytanhanig** [comment]
Some of us are still using Helm 3.6.* due to [bugs in Helm 3.7.0](https://github.com/helm/helm/issues/10149). If we do change the plugin name/helm command please ensure that there is a major increment in the plugin version to signal the introduction of a breaking change.

**Comment by eytanhanig** [comment]
I'm concerned that the Helm project released this breaking functionality despite chartmuseum not being updated to support the new standard and name change. `chartmuseum` is a popular, well-known, and widely adopted tool. What can we do to ensure that the `chartmuseum` and `helm` developers collaborate more closely so that this doesn't happen again?

**Comment by czenker** [comment]
A different drop-in-replacement for those willing to upgrade to Helm 3.7.0 is to use `helm-push`'s plugin binary directly.

Just replace

```
helm push ...
```

with 

```
${HOME}/.local/share/helm/plugins/helm-push/bin/helmpush ...
```

**Comment by gillesde** [comment]
Closing this issue as it was resolved in `chartmuseum/helm-push` version **v0.10.0**: https://github.com/chartmuseum/helm-push/releases/tag/v0.10.0

The solution is that from Helm v3.7.0 to be using the plugin **ChartMuseum Push** v0.10.0 used with: `helm cm-push`
